### PR TITLE
Upgrading to SpringBoot v3

### PR DESCRIPTION
**Pull request by meken**
Had to upgrade from javax to jakarta, bump the version numbers of a few libs, and fix a few tests (probably due to more strict JPA checks in the newest libs). Tested with JDK 17, also tested on Tomcat 10 (main reason for me to upgrade).

### CODE REVIEW

**Comment by arey** [comment]
Thanks @meken for tour contribution.
Could you please upgrade to Java 17 the `.github/workflows/maven-build.yml`?

File: src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataPetTypeRepositoryImpl.java [hunk 1]
```diff
@@ -41,8 +41,8 @@ public class SpringDataPetTypeRepositoryImpl implements PetTypeRepositoryOverrid
 	public void delete(PetType petType) {
         this.em.remove(this.em.contains(petType) ? petType : this.em.merge(petType));
 		Integer petTypeId = petType.getId();
-		
-		List<Pet> pets = this.em.createQuery("SELECT pet FROM Pet pet WHERE type_id=" + petTypeId).getResultList();
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataPetTypeRepositoryImpl.java:45]: Does this syntax not working any more?

File: src/main/java/org/springframework/samples/petclinic/security/BasicAuthenticationConfig.java [hunk 2]
```diff
@@ -10,31 +10,31 @@
...
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.crypto.factory.PasswordEncoderFactories;
 import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
 
 @Configuration
 @EnableGlobalMethodSecurity(prePostEnabled = true) // Enable @PreAuthorize method-level security
 @ConditionalOnProperty(name = "petclinic.security.enable", havingValue = "true")
-public class BasicAuthenticationConfig extends WebSecurityConfigurerAdapter {
```
Comment (arey) [inline src/main/java/org/springframework/samples/petclinic/security/BasicAuthenticationConfig.java:20]: Googbye `WebSecurityConfigurerAdapter` :)

File: src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataPetTypeRepositoryImpl.java (hunk 1 above)
Comment (meken) [reply src/main/java/org/springframework/samples/petclinic/repository/springdatajpa/SpringDataPetTypeRepositoryImpl.java:45]: Unfortunately that doesn't work anymore,  `ClinicServiceJPATests.shouldDeletePetType` fails with the exception: `org.hibernate.query.SemanticException: Could not interpret path expression 'type_id'`

File: src/main/java/org/springframework/samples/petclinic/security/BasicAuthenticationConfig.java (hunk 2 above)
Comment (meken) [reply src/main/java/org/springframework/samples/petclinic/security/BasicAuthenticationConfig.java:20]: Yeah, although other than the signature changes (both for the class & method), the logic is rather similar

**Pull request merged**